### PR TITLE
[Beam-566] Document the design proposal process

### PR DESCRIPTION
**Pull request by francesperry**
This is the second half of what was originally pull/42. 

Blocked on:
1. https://issues.apache.org/jira/browse/INFRA-12698
2. Updating/verifying JIRA links / names.

Reviewer: @davorbonaci

### CODE REVIEW

File: contribute/contribution-guide.md [hunk 1]
```diff
@@ -51,9 +51,42 @@ For moderate or large contributions, you should not start coding or writing a de
...
-Generally, the JIRA issue is the best place to gather relevant design docs, comments, or references. It’s great to explicitly include relevant stakeholders early in the conversation. For designs that may be generally interesting, we also encourage conversations on the developer’s mailing list.
+### Initiate a Proposal
+
+Large contributions should be tracked using the `proposal` type in JIRA. This allows the Beam community to quickly identify, track, and engage on fundamental changes to the Beam project.
+
+Proposals should be created for: 
+
+1. Changes to the Beam programming model, such as new primitive transformations or semantic changes to existing transformations.
+1. Significant changes to how Beam concepts are represented in a given SDK.
+1. Changes that may require coordinated effort across multiple components, such as shared API surfaces or testing infrastructure.
```
Comment (francesperry) [inline contribute/contribution-guide.md]: I agree major behavior changes to an existing runner belong here, as it affects existing users.

In my mind, I was thinking that creation of new components (runners, sdks, etc) would be handled by the feature branch process. That's a huge amount work, should be separable from what is happening in the core of the project, and won't affecting existing users or components. But it still has clear requirements for merging in to master. What do you think?

Comment (nssalian) [inline contribute/contribution-guide.md]: @francesperry , if it already has clear requirements elsewhere, I agree with your thought.

**Comment by nssalian** [comment]
LGTM, but I would like someone else to have more eyes on this. Thanks @francesperry  for putting this together.

**Comment by francesperry** [comment]
retest this please

**Comment by davorbonaci** [comment]
retest this pretty please

**Comment by davorbonaci** [comment]
retest this please

**Comment by davorbonaci** [comment]
retest this please

**Comment by davorbonaci** [comment]
retest this please

**Comment by davorbonaci** [comment]
retest this please

**Comment by davorbonaci** [comment]
retest this please

**Comment by nssalian** [comment]
may need a rebase @francesperry

**Comment by francesperry** [comment]
Yup -- will do at some point. This is still blocked on infra.

**Comment by iemejia** [comment]
This is the oldest PR for the beam site, is this still valid? any progress from the infra side ?

**Comment by davorbonaci** [comment]
Occasional progress... Freddy B. is on it, but not particularly prioritized. No ETA, as far as I know.

**Comment by jkff** [comment]
Doing a round of PR cleanup. This should probably be closed per the stale PR policy (60 days of inactivity) - please reopen if more progress is made.

**Pull request closed without merging**